CMakeLists.txt

```
@@ -8,8 +8,19 @@ if(POLICY CMP0072)
 	cmake_policy(SET CMP0072 NEW)
 	set(OpenGL_GL_PREFERENCE LEGACY)
 endif()
+
 project(mGBA)
-set(BINARY_NAME mgba CACHE INTERNAL "Name of output binaries")
+
+if(NOT DEFINED LIBMGBA_ONLY)
+	get_directory_property(LIBMGBA_ONLY EXCLUDE_FROM_ALL)
+endif()
+
+if(NOT LIBMGBA_ONLY)
+	set(BINARY_NAME mgba CACHE INTERNAL "Name of output binaries")
+elseif(NOT DEFINED BINARY_NAME)
+	set(BINARY_NAME mgba)
+endif()
```

I don't know if this would work but it seems like a better solution if it does, to avoid repetition.

Why the defined check, by the way?

```suggestion
if(NOT DEFINED BINARY_NAME)
	set(BINARY_NAME mgba)
endif()
if(NOT LIBMGBA_ONLY)
	set(BINARY_NAME ${BINARY_NAME} CACHE INTERNAL "Name of output binaries")
endif()
```

---

Avoiding repetition is indeed better, in that case the defined check is definitely necessary or it would overwrite the cache otherwise

---

Oh, yes, you're right.